doc/server_side_auth.md

```diff
@@ -2,7 +2,8 @@ Server-side API for Authenticating Clients
 ==========================================
 
 NOTE: This document describes how server-side authentication works in C-core based gRPC implementations only. In gRPC Java and Go, server side authentication is handled differently.
-NOTE2: `CallCredentials` class is only valid for secure channels in C-Core. So, for connections under insecure channels, features below might not be available.
+
+NOTE2: `CallCredentials` class is only valid if its security level is less than or equal to the security level of the associated connection. See the [gRFC](https://github.com/grpc/proposal/blob/master/L62-core-call-credential-security-level.md) for more.
```

Comment:
@yihuazhang would you mind taking a look at this here and see if the description make sense to you? Thank you so much!

Reply:
LGTM. But I will probably rephrase it as follows: "`CallCredentials` class is only valid if the security level it requires is less than or equal to the security level of connection used to transfer it. See the [gRFC](https://github.com/grpc/proposal/blob/master/L62-core-call-credential-security-level.md) for more information."

Reply:
Yeah that's much clearer. Thanks for the suggestion!